recipes/zstd/all/conanfile.py

```diff
@@ -51,41 +47,46 @@ def configure(self):
         del self.settings.compiler.cppstd
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["ZSTD_BUILD_PROGRAMS"] = False
-        self._cmake.definitions["ZSTD_BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["ZSTD_BUILD_SHARED"] = self.options.shared
-        self._cmake.definitions["ZSTD_MULTITHREAD_SUPPORT"] = self.options.threading
-        if tools.Version(self.version) < "1.4.3":
-            # Generate a relocatable shared lib on Macos
-            self._cmake.definitions["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        files.get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

Comment:
```suggestion
        files.get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

Already commented in another PR, but I think we should explicitly use a layout and refer to self.source_folder, self.build_folder etc afterwards, to make recipes consistent.

Reply:
another good reason to use `layout()`: https://github.com/conan-io/conan/issues/11867#issuecomment-1219300692